### ISSUE TITLE
Disable trace exporting via properties

### ISSUE THREAD

**Issue by kuldeep0508** [spring-projects/spring-boot#34620]
With spring boot 2.x  below mentioned dependency is required for Sleuth to send spans to Zipkin or a Zipkin compliant tracing distribution system.
`implementation 'org.springframework.cloud:spring-cloud-sleuth-zipkin'`

But with spring boot 3.x this `ZipkinAutoConfiguration` is deafult present in `org.springframework.boot.actuate.autoconfigure` and gets enabled when 

```properties
management.tracing.enabled = true
```
So MS start sending the spans. 

In our case we rely on istio for this task and our micro-services are not supposed to send any data to zipkin server. 

But with upgrade to springboot 3 this feature gets enabled as default and we get an error in our logs stating `unabled to connect with zipkin server` as we have not configure our MS to do this task. 

I check there is no property to control the enablement of this behaviour. 

So is there any plan to provide any properties like below to control this behavior. 

```properties
management.tracing.zipkin.enabled = true/false
```

And as a workaround should we write `noopZipkinSpanHandler` ?

**Comment by mhalbritter** [comment]
Does this help you? https://docs.spring.io/spring-boot/docs/current/reference/htmlsingle/#using.auto-configuration.disabling-specific

**Comment by kuldeep0508** [comment]
> Does this help you? https://docs.spring.io/spring-boot/docs/current/reference/htmlsingle/#using.auto-configuration.disabling-specific

It worked thanks :) , 

but  it may be good for a better control over it we can have a property for enable or disable it in future.

**Comment by nioertel** [comment]
I understand this argument for Spring Boot 3.3.x.

However with Spring Boot 3.2.x and before, Brave 5.x is used and there `brave.Span` directly depends on `zipkin2.Endpoint`. Removing Zipkin from the classpath is not a viable option.
Therefore I would not consider the presence of Zipkin on the classpath as sufficient signal.

**Comment by mhalbritter** [comment]
In that case, please use the auto-configuration disable feature, as I said [here](https://github.com/spring-projects/spring-boot/issues/34620#issuecomment-1471752928). We won't add an `enabled` flag in the 3.2.x line anyway, as it only gets bugfixes and dependency upgrades, but no new features.

**Comment by kubav182** [comment]
Why should we mess our config with autoconfig exclusions?
spring.zipkin.enabled - simple, clear and intuitive
spring.autoconfigure.exclude=org.springframework.boot.actuate.autoconfigure.tracing.zipkin.ZipkinAutoConfiguration why should i write this to config and search for this long classpath? IDE can help you but usually you have external config where is no autocomplete for spring boot packages. The user experience should be considered too. Spring Boot and its changes should make it easier not harder.

**Comment by rivancic** [comment]
OFC I have Zipkin on the classpath as application will use zipkin exporter when deployed in such environment. 
For local testing I don't need it. And like mentioned above, my first go to is to check Spring common application properties in how to enable/disable behavior. Now in this case you really have to know about the Autoconfiguration class.

**Comment by rishiraj88** [comment]
I fully agree with @rivancic . Autoconfiguration is so versatile and dependable.
We should always look up to this in first place while changing the inclusion/enabling of dependencies "automatically". It's such a powerful automation.

**Comment by kubav182** [comment]
Exactly we need to disable exporter for some reasons (tests, local env, maybe you dont need to export spans on every dev/test env). Removing zipkin from classpath is not option as we would have to make new build for each env. So having zipkin on classpath as signal is not the best option. Disabling autoconfiguration in yaml/properties is not user friendly from my point of view because we have to know all the autoconfigs relying on this signal instead of one shared enabled property. For example for redis there are 2 autoconfigs one relying on classpath and one relying on property it is quite confusing. I understand the argument there is complexity to have many ways to do the same thing, but our arguments are also valid I think.

**Comment by mhalbritter** [comment]
We talked about that and decided to add a `.enabled` property. This mirrors our metrics properties, e.g. there's a [`management.prometheus.metrics.export.enabled`](https://docs.spring.io/spring-boot/appendix/application-properties/index.html#application-properties.actuator.management.prometheus.metrics.export.enabled) property, or a [`management.graphite.metrics.export.enabled`](https://docs.spring.io/spring-boot/appendix/application-properties/index.html#application-properties.actuator.management.graphite.metrics.export.enabled). We want to do that for our tracing backends, too.

**Comment by Stonnos** [comment]
Workaround:
`
@Configuration
public class ZipkinReportAutoConfiguration {

    @Bean
    @ConditionalOnProperty(value = "management.zipkin.tracing.enabled", havingValue = "true")
    public AsyncReporter<Span> spanReporter(Sender sender, BytesEncoder<Span> encoder) {
        return AsyncReporter.builder(sender).build(encoder);
    }

    @Bean
    @ConditionalOnProperty(value = "management.zipkin.tracing.enabled", havingValue = "false", matchIfMissing = true)
    public Reporter<Span> noopReporter() {
        return span -> {
        };
    }
}
`

**Comment by mhalbritter** [comment]
Changes are here and ready to be merged as soon as we have our `3.3.x` branch: https://github.com/mhalbritter/spring-boot/tree/mh/34620-disable-trace-exporting-via-properties
